File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        ret = 0
        l = 0
        r = 0
        flips =1
        while(r<len(nums)):
            if nums[r] ==0:
                flips-=1    
            if flips<0:
                while flips<0:
                    if nums[l] == 0:
                        flips+=1
                    l+=1
            ret = max(ret,r-l)
            r+=1
        return ret

File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.ts
function longestSubarray(nums: number[]): number {
    let ret = 0;
    let l = 0
    let flip = 1
    for (let r = 0;r<nums.length;r++){
        if (nums[r]===0){
            flip--
        }   
        if (flip<0){
           
            while (flip<0){
                
                if (nums[l] ===0){
                    flip++
                }
                l++;
            }
        }
         ret = Math.max(r-l,ret)
    }
    return ret;   
};